src/actions/tests/pending-breakpoints.js

```diff
@@ -0,0 +1,261 @@
+      breakpointId2 = makePendingLocationId(breakpoint2.location);
+    });
+
+    it("adds a corresponding pendingBreakpoint for each new addition", async () => {
+      const { dispatch, getState } = createStore(simpleMockThreadClient);
+      await dispatch(actions.addBreakpoint(breakpoint1.location));
+      await dispatch(actions.addBreakpoint(breakpoint2.location));
+
+      const pendingBps = selectors.getPendingBreakpoints(getState());
+      expect(pendingBps.get(breakpointId1)).to.eql(
+        generatePendingBreakpoint(breakpoint1)
+      );
+      expect(pendingBps.get(breakpointId2)).to.eql(
+        generatePendingBreakpoint(breakpoint2)
+      );
+    });
+
+    it("removes a corresponding pending breakpoint for each deletion", async () => {
+      const { dispatch, getState } = createStore(simpleMockThreadClient);
+      await dispatch(actions.addBreakpoint(breakpoint1.location));
+      await dispatch(actions.addBreakpoint(breakpoint2.location));
+      await dispatch(actions.removeBreakpoint(breakpoint1.location));
+
+      const pendingBps = selectors.getPendingBreakpoints(getState());
+      expect(pendingBps.has(breakpointId1)).not.to.be(true);
+      expect(pendingBps.has(breakpointId2)).to.be(true);
+    });
+  });
+
+  it("when the user disables a breakpoint, the corresponding pending breakpoint is also disabled", async () => {
+    const { dispatch, getState } = createStore(simpleMockThreadClient);
+    const bp = generateBreakpoint("foo");
+    const id = makePendingLocationId(bp.location);
+
+    await dispatch(actions.addBreakpoint(bp.location));
+    await dispatch(actions.disableBreakpoint(bp.location));
+    const bps = selectors.getPendingBreakpoints(getState());
+    const breakpoint = bps.get(id);
+    expect(breakpoint.disabled).to.be(true);
+  });
+
+  it("when the user updates a breakpoint, the corresponding pending breakpoints is updated", async () => {
+    const { dispatch, getState } = createStore(simpleMockThreadClient);
+    const bp = generateBreakpoint("foo");
+    const id = makePendingLocationId(bp.location);
+
+    await dispatch(actions.addBreakpoint(bp.location));
+    await dispatch(
+      actions.setBreakpointCondition(bp.location, { condition: "2" })
+    );
+    const bps = selectors.getPendingBreakpoints(getState());
+    const breakpoint = bps.get(id);
+    expect(breakpoint.condition).to.be("2");
+  });
+
+  it("when the user updates a breakpoint, the corresponding pending breakpoints are not removed", async () => {
+    const { dispatch, getState } = createStore(simpleMockThreadClient);
+    const bp = generateBreakpoint("foo");
+    const id = makePendingLocationId(bp.location);
+
+    await dispatch(actions.addBreakpoint(bp.location));
+    await dispatch(
+      actions.setBreakpointCondition(bp.location, { condition: "2" })
+    );
+    const bps = selectors.getPendingBreakpoints(getState());
+    const breakpoint = bps.get(id);
+    expect(breakpoint.condition).to.be("2");
+  });
```

Comment:
proposed describe block `synced actions between breakpoints and pendingBreakpoints` should end here